libkbfs/disk_block_cache.go

```diff
@@ -879,12 +895,42 @@ func (cache *DiskBlockCacheLocal) evictFromTLFLocked(ctx context.Context,
 	return cache.evictSomeBlocks(ctx, numBlocks, blockIDs)
 }
 
-// evictLocked evicts a number of blocks from the cache.  We choose a pivot
-// variable b randomly. Then begin an iterator into cache.metaDb.Range(b,
-// MaxBlockID) and iterate from there to get numBlocks *
-// evictionConsiderationFactor block IDs.  We sort the resulting blocks by
-// value (LRU time) and pick the minimum numBlocks. We then call cache.Delete()
-// on that list of block IDs.
+func (cache *DiskBlockCacheLocal) randomTlfForEvictionLocked(minEvictBlocks int) (tlf.
+	ID, error) {
+	totalCount := 0
+	var priorityToEvict evictionPriority
+	for priorityToEvict = priorityNotHome; (totalCount <= minEvictBlocks) &&
+		(priorityToEvict <= priorityPrivateHome); priorityToEvict++ {
+		for tlfId, count := range cache.tlfCounts {
```

Comment:
This doesn't order them by ID, it's basically constructing your `priorityBlockCounts` struct each time in order to choose one to evict from below. But I agree that your strategy is better because it's more flexible and doesn't redo work.